firefox_puppeteer/tests/test_toolbars.py

```diff
@@ -28,6 +28,15 @@ def test_focus_and_clear(self):
         locationbar.clear()
         self.assertEqual(locationbar.value, '')
 
+    def test_gobutton(self):
+        # This replaces http://hg.mozilla.org/qa/mozmill-tests/file/default/firefox/tests/functional/testAwesomeBar/testGoButton.js
+        input_text = 'mozilla.org/'
+        locationbar = self.browser.navbar.locationbar
+        locationbar.urlbar.send_keys(input_text)
+        go_button = self.browser.navbar.go_button
+        self.assertEqual(locationbar.value, 'mozilla.org/')
+        self.assertFalse(go_button.is_displayed())
```

Comment:
The above lines do not match the test we have in Mozmill:

* it doesn't check that the button is hidden if no input has been given
* it does not check the functionality of the button by clicking it, that the target page has been loaded, and the button is hidden again